packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx

```diff
@@ -216,20 +238,28 @@ type CanvasInfo = {
   height: number;
   canvasWidth: number;
   canvasHeight: number;
+  proportion: number;
   viewport: PdfjsLib.PDFPageViewport;
 };
 
 function getCanvasInfo(
   loadedPage: PdfjsLib.PDFPageProxy | null | undefined,
-  scale: number
+  scale: number,
+  rootRef: RefObject<HTMLDivElement>
 ): CanvasInfo | null {
-  const canvasScale = window.devicePixelRatio ?? 1;
-  const viewport = loadedPage?.getViewport({ scale: scale * canvasScale });
-  if (viewport) {
+  const rootDimensions = rootRef?.current?.getBoundingClientRect();
+  if (loadedPage && rootDimensions) {
+    const canvasScale = window.devicePixelRatio ?? 1;
+    const width = rootDimensions.width * scale;
+    const pageWidth = loadedPage.view[2];
+    const proportion = rootDimensions.width / pageWidth;
+
+    const viewport = loadedPage.getViewport({
+      scale: (scale * canvasScale * width) / pageWidth
+    });
     const { width: canvasWidth, height: canvasHeight } = viewport;
-    const width = Math.ceil(canvasWidth / canvasScale);
-    const height = Math.ceil(canvasHeight / canvasScale);
-    return { width, height, canvasWidth, canvasHeight, viewport };
+    const height = (width * canvasHeight) / canvasWidth;
```

Comment:
Why is `height` calculated using width??

Reply:
The viewport gives us a height and width of the canvas. `canvasWidth / canvasHeight` gives us the height:width ratio for the specific document that we're looking at. Multiplying that by the display width gives us the display height that the document should have.

E.g. say we have a document that is 850px w x 1100px h (8.5" x 11" page), and we want to display it in a window of width 1000px. To determine the proper height, we would take 1100px (the PDF's actual height), divide by 850px (the PDF's actual width), then multiply by 1000px (the width of the space that we have to display it) = ~1294px for the display height.

In short, we need the width to compute the height because the size of the displayed PDF is entirely determined by the width now.

Reply:
Gotcha. Then clean up the code to make this evident. I would start by setting:

```js
const pageAspectRatio = canvasHeight / canvasWidth;
```